bpf/bpf_host.c

```diff
@@ -426,71 +531,66 @@ resolve_srcid_ipv4(struct __ctx_buff *ctx, __u32 srcid_from_proxy,
 	return src_id;
 }
 
+struct {
+	__uint(type, BPF_MAP_TYPE_PERCPU_ARRAY);
+	__type(key, __u32);
+	__type(value, struct ct_buffer4);
+	__uint(max_entries, 1);
+} CT_TAIL_CALL_BUFFER4 __section_maps_btf;
+
 static __always_inline int
-handle_ipv4(struct __ctx_buff *ctx, __u32 secctx,
-	    __u32 ipcache_srcid __maybe_unused, const bool from_host, __s8 *ext_err __maybe_unused)
+handle_ipv4_cont(struct __ctx_buff *ctx, __u32 secctx,
+		 __s8 *ext_err __maybe_unused)
 {
 	struct trace_ctx __maybe_unused trace = {
 		.reason = TRACE_REASON_UNKNOWN,
 		.monitor = TRACE_PAYLOAD_LEN,
 	};
-	struct remote_endpoint_info *info = NULL;
-	__u32 __maybe_unused remote_id = 0;
-	struct endpoint_info *ep;
+	__u32 from_host_raw = ctx_load_meta(ctx, CB_FROM_HOST);
```

Comment:
One thing of note here is that in current code, `from_host` is a constant. So the compiler can effectively optimize ~half of the code out. Now we need to include support for both cases ... that doesn't feel like a win, complexity-wise?